test/run_test.py

```diff
@@ -1123,22 +1158,54 @@ def main():
         # downloading test cases configuration to local environment
         get_test_case_configs(dirpath=test_directory)
 
-    has_failed = False
     failure_messages = []
+
+    selected_tests_parallel = [x for x in selected_tests if not must_serial(x)]
+    selected_tests_serial = [x for x in selected_tests if x not in selected_tests_parallel]
+    print_to_stderr("parallel tests:\n {}".format("\n ".join(selected_tests_parallel)))
```

Comment:
Just a note: this might be a bit misleading to read given what we discussed earlier:

* `parallel tests` means that these test files can be run in parallel
* `serial tests` means that these test files needs to run sequentially, but they could use multiple shards to run multiple process "in parallel"

Reply:
added a comment, also changed the print statements slightly but idk if its much clearer